src/main/java/seedu/address/logic/parser/DateTimeParser.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.reminder.DateTime;
+
+/**
+ * Contains utility methods used for parsing date time strings.
+ */
+public class DateTimeParser {
+    public static final String DATE_TIME_FORMATTER = "yyyy-MM-dd 'at' HH:mm";
+    public static final String[] VALID_DATE_24HOUR_TIME_FORMATS = {
+        "yy-M-d H:mm", "yy-MMM-d H:mm", "yyyy-M-d H:mm", "yyyy-MMM-d H:mm"};
+    public static final String[] VALID_DATE_AMPM_TIME_FORMATS = {
+        "yy-M-d h:mm a", "yy-MMM-d h:mm a", "yyyy-M-d h:mm a", "yyyy-MMM-d h:mm a",
+        "yy-M-d h:mma", "yy-MMM-d h:mma", "yyyy-M-d h:mma", "yyyy-MMM-d h:mma"};
+    public static final String[][] ALL_DATE_TIME_FORMATS = {
+        VALID_DATE_24HOUR_TIME_FORMATS, VALID_DATE_AMPM_TIME_FORMATS };
+
+    public static final String MESSAGE_INVALID_DATE_TIME_FORMAT = "Please enter a valid date time "
+            + "in the following format: year-month-day hour:minute\n"
+            + "Example: 2022-12-27 23:00, 22-08-25 07:30, 22-nov-11 7:00PM";
+
+    /**
+     * Parses a {@code String dateTime} into a {@code DateTime}.
+     *
+     * @throws ParseException if the given {@code String dateTime} is invalid.
+     */
+    public static DateTime parseDateTime(String dateTimeString) throws ParseException {
+
+        requireNonNull(dateTimeString);
+
+        LocalDateTime dateTime = null;
+
+        try {
+            dateTime = LocalDateTime.parse(dateTimeString);
+        } catch (DateTimeParseException e) {
+            boolean isValidDateTime = false;
+            for (int i = 0; i < ALL_DATE_TIME_FORMATS.length; i++) {
+                dateTime = parseLocalDateTime(dateTimeString, ALL_DATE_TIME_FORMATS[i]);
+                if (dateTime != null) {
+                    isValidDateTime = true;
+                    break;
+                }
+            }
+            if (!isValidDateTime) {
+                throw new ParseException(MESSAGE_INVALID_DATE_TIME_FORMAT);
+            }
+        }
```

Comment:
i presume this is to check if the datetimestring is in one of the listed formats. Why didn't you use the DateTime.isValidDateTime() method?

Reply:
It's because `DateTime` object only checks `yyyy-MM-dd 'at' HH:mm` which is used only used for validation creation of the `DateTime` object (i.e. just a defensive programming mechanism), whereas the parser here validates and converts the String from many possible formats that are entered by the user.

Reply:
@weiquany Though I want to add that if you're not using the `DateTime.isValidDateTime()` method outside of the `DateTime` code (didn't check whether this is the case myself), you should make it private.

Reply:
I understand what you are doing now. Maybe `DateTime.isValidDateTime()` can accept another argument for the datetime format to check with? Also, would making `ALL_DATE_TIME_FORMATS` a 1D array make things simpler? 
```java
String[] ALL_DATE_TIME_FORMATS = ArrayUtils.addAll(VALID_DATE_24HOUR_TIME_FORMATS, VALID_DATE_AMPM_TIME_FORMATS);
```

Reply:
@malwaregarry Thanks for the suggestions. I simplified things a bit. Please take a look at it. Can you clarify what you meant by your first statement? Currently, there is only a `DateTime.isValidDateTimeString()`. The underlying structure uses a `LocalDateTime` object, which does not require verification.

Reply:
@weiquany the method currently uses the `DATE_TIME_FORMATTER` variable from the class. Instead of relying on it, what I suggested was to allow a format to be passed in
```java
    public static boolean isValidDateTimeString(String dateTimeString, String dateTimeFormat) {
        try {
            LocalDateTime.parse(dateTimeString, DateTimeFormatter.ofPattern(dateTimeFormat));
            return true;
        } catch (DateTimeParseException pe) {
            return false;
        }
    }
```

Reply:
Thanks for the suggestion @malwaregarry